packages/remirror__extension-list/src/list-commands.ts

```diff
@@ -151,20 +147,29 @@ export function splitListItem(listItemType: string | NodeType): CommandFunction
     }
 
     const nextType = $to.pos === $from.end() ? grandParent.contentMatchAt(0).defaultType : null;
-    tr.delete($from.pos, $to.pos);
 
-    const types: TypesAfter = [undefined];
+    const newTypeAttributes = Object.fromEntries(
+      Object.entries(grandParent.attrs).filter(([attr]) => !ignoreAttrs.includes(attr)),
+    );
 
-    if (nextType) {
-      types.push({ type: nextType, attrs: {} });
-    }
+    const newNextTypeAttributes = { ...$from.node().attrs };
+
+    tr.delete($from.pos, $to.pos);
+
+    const types: TypesAfter = nextType
+      ? [
+          { type, attrs: newTypeAttributes },
+          { type: nextType, attrs: newNextTypeAttributes },
+        ]
+      : [{ type, attrs: newTypeAttributes }];
 
-    if (!canSplit(tr.doc, $from.pos, 2, types)) {
+    if (!canSplit(tr.doc, $from.pos, 2)) {
+      // I can't use `canSplit(tr.doc, $from.pos, 2, types)` and I don't know why
```

Comment:
@ifiokjr  Here is another place I found strange. `canSplit` accepts an optional parameter `type`, but it will always return `false` if we add `types` here. I don't know why but I decided to remove `types` for now since Tiptap doesn't provide `types` here.